controller/src/main/java/io/pravega/controller/store/stream/AbstractStreamMetadataStore.java

```diff
@@ -76,6 +80,106 @@
     private static final OpStatsLogger DELETE_STREAM = STATS_LOGGER.createStats(MetricsNames.DELETE_STREAM);
     private final static String RESOURCE_PART_SEPARATOR = "_%_";
 
+    @Data
+    /**
+     * This class represents a 96 bit number with 32 bit msb encoded as integer and 64 bit lsb encoded as long.
+     * It is unsigned and only allows for positive values. It also implements comparable interface and comparison involves
+     * first compariging msbs and if msbs are equal then we compare lsbs.
+     */
+    protected static class BigLong implements Comparable {
+        public static final BigLong ZERO = new BigLong(0, 0L);
+        private final int msb;
+        private final long lsb;
+
+        BigLong(int msb, long lsb) {
+            Preconditions.checkArgument(msb >= 0);
+            Preconditions.checkArgument(lsb >= 0);
+
+            this.msb = msb;
+            this.lsb = lsb;
+        }
+
+        @Override
+        public int compareTo(Object o) {
+            if (!(o instanceof BigLong)) {
+                throw new RuntimeException("incomparable objects");
+            }
+            BigLong other = (BigLong) o;
+
+            if (msb != other.msb) {
+                return Integer.compare(msb, other.msb);
+            } else {
+                return Long.compare(lsb, other.lsb);
+            }
+        }
+
+        byte[] toBytes() {
```

Comment:
I didnt think it necessary because its a 96 bit number we are creating. 
there were three ways in which we could have composed it -- 3 ints, int + long or long + int. 
all are equivalent 96 bits at the end of the day.